### PR TITLE
Support for ExternalVR implementation

### PR DESCRIPTION
**Pull request by paulrouget**
This PR adds the hook necessary for the ExternalVR rust-webvr driver.
Waiting on rust-webvr 0.9.3 to be published before landing.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/22773)

### CODE REVIEW

File: ports/libsimpleservo/jniapi/src/lib.rs [hunk 1]
```diff
@@ -686,13 +688,21 @@ fn get_options(env: &JNIEnv, opts: JObject) -> Result<(InitOptions, bool, Option
         get_non_null_field(env, opts, "enableSubpixelTextAntialiasing", "Z")?
             .z()
             .map_err(|_| "enableSubpixelTextAntialiasing not a boolean")?;
+    let vr_pointer = get_non_null_field(env, opts, "VRExternalContext", "J")?
+        .j()
+        .map_err(|_| "height not an int")? as *mut c_void;
```
Comment (jdm) [inline ports/libsimpleservo/jniapi/src/lib.rs]: Wrong error message.

**Review by jdm — CHANGES_REQUESTED** [review]

**Comment by jdm** [comment]
Appveyor:
```
error[E0432]: unresolved import `rust_webvr::api::VRExternalShmemPtr`
  --> components\webvr\lib.rs:12:9
   |
12 | pub use rust_webvr::api::VRExternalShmemPtr;
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `VRExternalShmemPtr` in `api`
error[E0599]: no method named `register_vrexternal` found for type `rust_webvr::vr_manager::VRServiceManager` in the current scope
  --> components\webvr\webvr_thread.rs:61:21
   |
61 |             service.register_vrexternal(ptr);
   |                     ^^^^^^^^^^^^^^^^^^^
error: aborting due to 2 previous errors
```
Travis is extremely displeased by the rust-webvr upgrade.

**Comment by paulrouget** [comment]
That fails because of Servo, not webvr. I'm going to make it so webvr exposes the same API no matter the platform: https://github.com/servo/rust-webvr/pull/44

**Comment by paulrouget** [comment]
Fixed non-android builds on rust-webvr side.
Fixed the error message.

@bors-servo r=jdm

**Comment by jdm** [comment]
```
error[E0560]: struct `api::vrexternal::android::mozgfx::VRLayer_Stereo_Immersive` has no field named `__bindgen_padding_0`
   --> /Users/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/rust-webvr-0.9.15/src/api/vrexternal/android/display.rs:227:17
    |
227 |                 __bindgen_padding_0: 0,
    |                 ^^^^^^^^^^^^^^^^^^^ `api::vrexternal::android::mozgfx::VRLayer_Stereo_Immersive` does not have this field
    |
    = note: available fields are: `mTextureHandle`, `mTextureType`, `mFrameId`, `mInputFrameId`, `mLeftEyeRect`, `mRightEyeRect`

error[E0560]: struct `api::vrexternal::android::mozgfx::VRLayerState` has no field named `__bindgen_padding_0`
   --> /Users/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/rust-webvr-0.9.15/src/api/vrexternal/android/display.rs:234:13
    |
234 |             __bindgen_padding_0: 0,
    |             ^^^^^^^^^^^^^^^^^^^ `api::vrexternal::android::mozgfx::VRLayerState` does not have this field
    |
    = note: available fields are: `type_`, `__bindgen_anon_1`

error: aborting due to 2 previous errors
```
The way we solve this in other places is:
```rust
        let layer_stereo_immersive = {
            let rendered_layer = self.rendered_layer.as_ref().unwrap();
            mozgfx::VRLayer_Stereo_Immersive {
                mTextureHandle: rendered_layer.texture_id as u64,
                mTextureType: mozgfx::VRLayerTextureType_LayerTextureType_GeckoSurfaceTexture,
                mFrameId: self.system_state.sensorState.inputFrameID,
                mLeftEyeRect: mozgfx::VRLayerEyeRect {
                    x: rendered_layer.left_bounds[0],
                    y: rendered_layer.left_bounds[1],
                    width: rendered_layer.left_bounds[2],
                    height: rendered_layer.left_bounds[3],
                },
                mRightEyeRect: mozgfx::VRLayerEyeRect {
                    x: rendered_layer.right_bounds[0],
                    y: rendered_layer.right_bounds[1],
                    width: rendered_layer.right_bounds[2],
                    height: rendered_layer.right_bounds[3],
                },
                mInputFrameId: 0,
                ...mozgfx::VRLayer_Stereo_Immersive::default()
            }
};
```

**Comment by paulrouget** [comment]
Android x86 build error.

**Comment by paulrouget** [comment]
Hopefully this is the good one.

@bors-servo r=jdm

**Pull request merged**